Mathlib/AlgebraicTopology/SimplicialSet/StrictSegal.lean

```diff
@@ -0,0 +1,156 @@
+/-
+Copyright (c) 2024 Emily Riehl. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Mario Carneiro, Emily Riehl, Joël Riou
+-/
+import Mathlib.AlgebraicTopology.SimplicialSet.Nerve
+import Mathlib.AlgebraicTopology.SimplicialSet.Path
+import Mathlib.CategoryTheory.Functor.KanExtension.Adjunction
+import Mathlib.CategoryTheory.Functor.KanExtension.Basic
+
+/-!
+# Strict Segal simplicial sets
+
+A simplicial set `X` satisfies the `StrictSegal` condition if for all `n`, the map
+`X.spine n : X _[n] → X.Path n` is a bijection.
+
+Examples of `StrictSegal` simplicial sets are given by nerves of categories.
+
+TODO: Show that these are the only examples: that a `StrictSegal` simplicial set is isomorphic to
+the nerve of its homotopy category.
+
+`StrictSegal` simplicial sets have an important property of being 2-coskeletal which is proven
+elsewhere.
+-/
+
+universe v u
+
+open CategoryTheory
+
+open Simplicial SimplexCategory
+
+namespace SSet
+
+variable (X : SSet.{u})
+
+/-- A simplicial set `X` satisfies the strict Segal condition if its simplices are uniquely
+determined by their spine. -/
+class StrictSegal : Prop where
+  segal : ∀ (n : ℕ), Function.Bijective (X.spine n)
```

Comment:
I don't have experience with this concept. Do you think it will be useful to make the inverse function (ie `spineToSimplex`) part of this class? Then the inverse would be computable, so people can play with defeqs.

But it also means that this is no longer a property, but data-carrying.

I have zero opinion on this, just wanted to hear your thoughts.

Reply:
Thanks. This was a great suggestion. The new code seems much cleaner to me. Let me know if I've implemented everything correctly.